common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -140,6 +148,87 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         }
     }
 
+    @Override
+    public void onResume() {
+        super.onResume();
+        final String methodTag = TAG + ":onResume";
+        final Bundle extras = getExtras();
+        if (isSwitchBrowserResumeFlow(extras)) {
+            Logger.info(methodTag, "Resume switch browser protocol on WebView.");
+            resumeSwitchBrowser(extras);
+        }
+    }
+
+    /**
+     * Get the extras from the activity intent.
+     *
+     * @return Bundle with the extras
+     */
+    @NonNull
+    private Bundle getExtras() {
+        final Activity activity = getActivity();
+        if (activity == null) {
+            return Bundle.EMPTY;
+        }
+        final Intent intent = activity.getIntent();
+        if (intent == null) {
+            return Bundle.EMPTY;
+        }
+        final Bundle extras =  intent.getExtras();
+        return extras == null ? Bundle.EMPTY : extras;
+    }
+
+    /**
+     * Check if the extras contains the switch browser code and action uri.
+     * also checks that the fragment is waiting for the switch browser resume endpoint.
+     * if so, it means we are resuming the switch browser flow.
+     *
+     * @param extras Bundle
+     * @return boolean
+     */
+    private boolean isSwitchBrowserResumeFlow(@NonNull final Bundle extras) {
+        final String methodTag = TAG + ":isSwitchBrowserResumeFlow";
+        final boolean containsActionUri = extras.containsKey(
+                AuthenticationConstants.SWITCH_BROWSER.ACTION_URI
+        );
+        final boolean containsCode = extras.containsKey(
+                AuthenticationConstants.SWITCH_BROWSER.CODE
+        );
+        Logger.verbose(
+                methodTag,
+                "action uri: " + containsActionUri +
+                        ", code: " + containsCode +
+                        ", isWaitingForSwitchBrowserResumeEndpoint: " + isWaitingForSwitchBrowserResumeEndpoint);
+        return containsCode && containsActionUri && isWaitingForSwitchBrowserResumeEndpoint;
+    }
+
+    /**
+     * Resume the switch browser protocol flow.
+     *
+     * @param extras Bundle with the data to resume the switch browser protocol flow.
+     */
+    private void resumeSwitchBrowser(@NonNull final Bundle extras) {
+        final String methodTag = TAG + ":resumeSwitchBrowser";
+        final String actionUri = extras.getString(
+                AuthenticationConstants.SWITCH_BROWSER.ACTION_URI
+        );
+        final String code = extras.getString(
+                AuthenticationConstants.SWITCH_BROWSER.CODE
+        );
+        if (actionUri == null || code == null) {
+            Logger.error(methodTag, "Action URI is null: " + (actionUri == null) +
+                    ", code is null: " + (code == null), null);
+            cancelAuthorization(false);
```

Comment:
How is this error surfaced back to client? Does this come back as User cancel?

We need to send this back as cancel with some error information about the required fields being missing and also need to report this in telemetry

Reply:
Reworked and call sendResult directly to surface the exception to the client.
regarding telemetry I will have my final PR with the telemetry :)